src/app/clusters/temperature-measurement-server/temperature-measurement-server.cpp

```diff
@@ -15,78 +15,88 @@
-// clear where best to declare it.
-// https://github.com/project-chip/connectedhomeip/issues/3619
-void emberAfPluginTemperatureMeasurementServerReadEventHandler() {}
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMeasuredValue(chip::EndpointId endpoint, int16_t * measuredValue)
+{
+    return emberAfReadServerAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MEASURED_VALUE_ATTRIBUTE_ID,
+                                      (uint8_t *) measuredValue, sizeof(*measuredValue));
+}
 
-void emberAfPluginTemperatureMeasurementServerStackStatusCallback(EmberStatus status) {}
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMinMeasuredValue(chip::EndpointId endpoint, int16_t * minMeasuredValue)
+{
+    return emberAfReadServerAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MIN_MEASURED_VALUE_ATTRIBUTE_ID,
+                                      (uint8_t *) minMeasuredValue, sizeof(*minMeasuredValue));
+}
 
-// -------------------------------------------------------------------------
-// ****** callback section *******
+EmberAfStatus emberAfTemperatureMeasurementClusterGetMaxMeasuredValue(chip::EndpointId endpoint, int16_t * maxMeasuredValue)
+{
+    return emberAfReadServerAttribute(endpoint, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_TEMP_MAX_MEASURED_VALUE_ATTRIBUTE_ID,
+                                      (uint8_t *) maxMeasuredValue, sizeof(*maxMeasuredValue));
+}
 
-void emberAfPluginTemperatureMeasurementServerInitCallback(void)
+void emberAfTemperatureMeasurementClusterServerInitCallback(chip::EndpointId endpoint)
 {
-    EmberAfStatus status;
-    // FIXME Use real values for the temperature sensor polling the sensor using the
-    //       EMBER_AF_PLUGIN_TEMPERATURE_MEASUREMENT_SERVER_MAX_MEASUREMENT_FREQUENCY_S macro
-    EndpointId endpointId = 1; // Hardcoded to 1 for now
-    int16_t newValue      = 0x1234;
+    int16_t value        = 0;
+    EmberAfStatus status = emberAfTemperatureMeasurementClusterGetMeasuredValue(endpoint, &value);
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
+    {
+        status = emberAfTemperatureMeasurementClusterSetMeasuredValueCallback(endpoint, INT16_MIN);
+        if (EMBER_ZCL_STATUS_SUCCESS != status)
+        {
+            emberAfTemperatureMeasurementClusterPrintln("ERR: writing present value %x", status);
+        }
+    }
 
-    status = emberAfWriteAttribute(endpointId, ZCL_TEMP_MEASUREMENT_CLUSTER_ID, ZCL_CURRENT_TEMPERATURE_ATTRIBUTE_ID,
-                                   CLUSTER_MASK_SERVER, (uint8_t *) &newValue, ZCL_INT16S_ATTRIBUTE_TYPE);
-    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    status = emberAfTemperatureMeasurementClusterGetMinMeasuredValue(endpoint, &value);
+    if (EMBER_ZCL_STATUS_SUCCESS != status)
     {
-        emberAfTempMeasurementClusterPrint("Err: writing temperature: %x", status);
-        return;
+        status = emberAfTemperatureMeasurementClusterSetMinMeasuredValueCallback(endpoint, INT16_MIN);
```

Comment:
Shouldn't we set the minimum value to INT16_MAX?

Reply:
Ah, it's a special value. Could we have a constant "kUnknownTemperatureValue = INT16_MIN" defined at the beginning of the file?

Reply:
Btw, I can see that `all-clusters-app.zap`  has `"defaultValue": "0x8000"`. Does it have to be initialized in this callback then?

Reply:
It should be already stored into the attribute store.

Reply:
I have removed the default impl. in the init function. Because it is like @vivien-apple mentioned already stored in the attribte storage.